src/utils/StorageHandler.ts

```diff
@@ -547,12 +547,19 @@ export default class StorageHandler {
         );
       }
     } else {
-      const allTargetApps = await this.getTargetAppNames();
+      const impactedTargetApps = new Set(entity.targetApps);
+      targetApps.forEach(value => impactedTargetApps.add(value));
```

Comment:
Previously we only iterated on targetApps already registered in statsig on prem. Are there any concerns about unregistered targetApps being passed in here?

Reply:
Good question, I suppose we could resole that by still querying all target apps and then getting the union of all target apps and the set of affected target apps?

Reply:
should've been intersection but I ended up going with an easier route:
1. assume that any already saved targetApps on a gate are real existing target apps
2. when iterating through the new targetApp args check if they exist before adding them to the set